From fc7634b4fb644d29e55f59e0074dc78b2ffc878e Mon Sep 17 00:00:00 2001
From: Geetha sowjanya <gakula@caviumnetworks.com>
Date: Mon, 4 Feb 2019 17:20:43 +0530
Subject: [PATCH 0935/1051] soc: octeontx2: Add PTP subsystem ID for 95xx chip

The PTP subsystem ID are different for 96xx and 95xx chips.
Adding 95xx PTP subsystem ID to get PTP driver loaded.

Change-Id: I81112d9de34e65834fbc7b2b35c39f9ad234812d
Signed-off-by: Geetha sowjanya <gakula@caviumnetworks.com>
Reviewed-on: https://sj1git1.cavium.com/3408
Tested-by: sa_ip-sw-jenkins
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/soc/marvell/octeontx2/ptp.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/drivers/soc/marvell/octeontx2/ptp.c b/drivers/soc/marvell/octeontx2/ptp.c
index 56bc8aad8302..07ecf6895e94 100644
--- a/drivers/soc/marvell/octeontx2/ptp.c
+++ b/drivers/soc/marvell/octeontx2/ptp.c
@@ -17,7 +17,8 @@
 #define DRV_NAME	"Marvell PTP Driver"
 
 #define PCI_DEVID_OCTEONTX2_PTP		0xA00C
-#define PCI_SUBSYS_DEVID_OCTEONTX2_PTP	0xB200
+#define PCI_SUBSYS_DEVID_OCTX2_96XX_PTP	0xB200
+#define PCI_SUBSYS_DEVID_OCTX2_95XX_PTP	0xB300
 #define PCI_DEVID_OCTEONTX2_RST		0xA00E
 
 #define PCI_PTP_BAR_NO	0
@@ -199,7 +200,10 @@ static void ptp_remove(struct pci_dev *pdev)
 static const struct pci_device_id ptp_id_table[] = {
 	{ PCI_DEVICE_SUB(PCI_VENDOR_ID_CAVIUM, PCI_DEVID_OCTEONTX2_PTP,
 			 PCI_VENDOR_ID_CAVIUM,
-			 PCI_SUBSYS_DEVID_OCTEONTX2_PTP) },
+			 PCI_SUBSYS_DEVID_OCTX2_96XX_PTP) },
+	{ PCI_DEVICE_SUB(PCI_VENDOR_ID_CAVIUM, PCI_DEVID_OCTEONTX2_PTP,
+			 PCI_VENDOR_ID_CAVIUM,
+			 PCI_SUBSYS_DEVID_OCTX2_95XX_PTP) },
 	{ 0, }
 };
 
-- 
2.17.1

